tests/src/test/scala/cats/tests/TransLiftTests.scala

```diff
@@ -0,0 +1,42 @@
+package cats
+package tests
+
+import data.{OptionT,XorT,WriterT,StreamingT, Kleisli, StateT}
+
+class TransLiftTests extends CatsSuite {
+
+  case class NoTypeclass[A](a: A)
+
+  case class JustFunctor[A](a: A)
+
+  implicit val jfFunctor: Functor[JustFunctor] = new Functor[JustFunctor] {
+    override def map[A,B](fa: JustFunctor[A])(f: A => B): JustFunctor[B] = JustFunctor(f(fa.a))
+  }
+
+  case class JustAp[A](a: A)
+
+  implicit val jfApp: Applicative[JustAp] = new Applicative[JustAp] {
+    override def pure[A](a: A): JustAp[A] = JustAp(a)
+    override def ap[A, B](ff: JustAp[A => B])(fa: JustAp[A]): JustAp[B] = JustAp(ff.a(fa.a))
+    override def product[A, B](fa: JustAp[A],fb: JustAp[B]): JustAp[(A, B)] = JustAp(fa.a -> fb.a)
+    override def map[A, B](fa: JustAp[A])(f: A => B): JustAp[B] = JustAp(f(fa.a))
+  }
+
+  test("transLift for XorT, OptionT, WriterT requires only Functor") {
+    val d: XorT[JustFunctor, Int, Int]    = JustFunctor(1).liftT[({type λ[α[_], β] = XorT[α, Int, β]})#λ]
+    val c: OptionT[JustFunctor, Int]      = JustFunctor(1).liftT[OptionT]
+    val a: WriterT[JustFunctor, Int, Int] = JustFunctor(1).liftT[({type λ[α[_], β] = WriterT[α, Int, β]})#λ]
+
+  }
+
+  test("transLift for StreamingT, StateT require Applicative Functor") {
+    import StreamingT._
+    
+    val b: StreamingT[JustAp, Int] = JustAp(1).liftT[StreamingT]
+    val f: StateT[JustAp, Int, Int]  = JustAp(1).liftT[({type λ[α[_], β] = StateT[α, Int, β]})#λ]
+  }
+
+  test("transLift for, Kleisli doesn't require anything of the wrapped value"){
+    val e: Kleisli[NoTypeclass, Int, Int] = NoTypeclass(1).liftT[({type λ[α[_], β] = Kleisli[α, Int, β]})#λ]
+  }
+}
```

Comment:
I think it would be good to have a test to see if we can lift a `ValidatedNel` into some transformers. I suspect it would reveal that we currently can't use syntax to do that because we don't have `Unapply`-based syntax.